src/axom/core/MDMapping.hpp

```diff
@@ -106,23 +107,42 @@ class ArrayIndexer
 
     Object must be initialized before use.
   */
-  ArrayIndexer() = default;
+  MDMapping() = default;
+
+  /*!
+    @brief Construct mapping for empty shape and the given stride order.
+
+    The expected use case is when a stride order is known but the
+    shape is to be determined.  Initialize the mapping with its own
+    slowestDirs() to preserve stride order as its shape changes.
+  */
+  MDMapping(ArrayStrideOrder arrayStrideOrder)
+  {
+    axom::StackArray<T, DIM> shape;
+    for(int d = 0; d < DIM; ++d)
+    {
+      shape[d] = 0;
+    }
+    initializeShape(shape, arrayStrideOrder);
+  }
 
   /*!
     @brief Initialize for row- or column-major indexing.
     @param [in] shape Shape of the array
-    @param [in] arrayStrideOrder An order indicator from
-                ArrayStrideOrder.
+    @param [in] arrayStrideOrder An order indicator,
+      not ArrayStrideOrder::ARBITRARY.
     @param [in] fastestStrideLength Stride in the fastest
-                direction.
+                chnaging direction.
```

Comment:
typo: correct to "fastest-changing"

Reply:
Please change Doxygen for all other occurrences of `fastestStrideLength` throughout this PR to say "fastest-changing direction"

Reply:
Done.